arangod/Agency/Constituent.cpp

```diff
@@ -227,7 +237,7 @@ void Constituent::lead(term_t term) {
 
     // if we already have a higher term, ignore this request
     if (term < _term) {
-      followNoLock(_term);
+      followNoLock(0);  // leave _term and _votedFor unchanged
```

Comment:
compile error?  followNoLock(0, NO_LEADER)?

Reply:
We went back and forth between removing the term 0 case for not changing things and in the end we left it there, and made it so that the _votedFor is unchanged too in this case. Therefore it looks nicer to have the default argument with `_votedFor = NO_LEADER`. This is a matter of taste.